noria-server/src/controller/sql/mir/join.rs

```diff
@@ -46,24 +48,31 @@ pub fn make_joins(
 
     for jref in qg.join_order.iter() {
         let (join_type, jp) = from_join_ref(jref, &qg);
-        let (left_chain, right_chain) =
+        let (left_chain, second_chain) =
             pick_join_chains(&jref.src, &jref.dst, &mut join_chains, node_for_rel);
 
-        let jn = mir_converter.make_join_node(
-            &format!("{}_n{}", name, node_count),
-            jp,
-            left_chain.last_node.clone(),
-            right_chain.last_node.clone(),
-            join_type,
-        );
+        match second_chain {
+            Some(right_chain) => {
+                let jn = mir_converter.make_join_node(
+                    &format!("{}_n{}", name, node_count),
+                    jp,
+                    left_chain.last_node.clone(),
+                    right_chain.last_node.clone(),
+                    join_type,
+                );
 
-        // merge node chains
-        let new_chain = left_chain.merge_chain(right_chain, jn.clone());
-        join_chains.push(new_chain);
+                // merge node chains
+                let new_chain = left_chain.merge_chain(right_chain, jn.clone());
+                join_chains.push(new_chain);
 
-        node_count += 1;
+                node_count += 1;
 
-        join_nodes.push(jn);
+                join_nodes.push(jn);
+            },
+            None => {
+                join_chains.push(left_chain);
```

Comment:
Putting back `left_chain` in this case is definitely more correct than what we had before, but I wonder if we need to be careful about where to put it in the vector.

`pick_join_chains` searches the vector of chains in order, so we'll pull out the first chain that has all the tables we're looking for. That may be a longer chain than we're looking for (e.g., we might find ABC when a later entry in the vector has AB). I _think_ this is fine because later code will scan along the longer chain and find the AB join without caring about the extra C, but can you double-check that this is the case?